src/ChunkDef.h

```diff
@@ -31,7 +31,13 @@ class cChunkCoords;
 
 typedef std::unique_ptr<cEntity> OwnedEntity;
 typedef std::vector<OwnedEntity> cEntityList;
-typedef std::map<int, cBlockEntity *> cBlockEntities;
+
+struct sBlockEntityDeleter
+{
+	void operator () (cBlockEntity * a_BlockEntity) const NOEXCEPT;
+};
```

Comment:
Why exactly do we need this custom deleter, that does exactly the same as the regular deleter?

I assume it's the same issue I've come across with `unique_ptr`: That classes having them as members then need their destructor defined in a place where the pointed-to object is defined as well. This is usually better solved by providing an empty destructor in the class's cpp file, rather than having a custom deleter.

Reply:
I thought you might be able to shed some light, the original sBlockEntityDeleter did appear in BlockArea code XD (line 2733)

Reply:
`delete be;` requires a full definition of `cBlockEntity` whereas `sBlockEntityDeleter{}(be)` only requires `cBlockEntity` be forward declared.

Reply:
Yes, but on the other hand it keeps us from using `make_unique` and requires the conversions between `unique_ptr` and `OwnedBlockEntity`.

Reply:
@peterbell10 If you used a regular default-deleter `unique_ptr`, how much would it affect the rest of the code? How many places (order of 1, 10, 100?) would need changing?